app/src/main/java/com/arygm/quickfix/model/category/CategoryRepositoryFirestore.kt

```diff
@@ -23,60 +27,83 @@ open class CategoryRepositoryFirestore(private val db: FirebaseFirestore) : Cate
     }
   }
 
-  override fun fetchCategories(
+  override suspend fun fetchCategories(
       onSuccess: (List<Category?>) -> Unit,
       onFailure: (Exception) -> Unit
   ) {
-    CoroutineScope(Dispatchers.IO).launch {
-      try {
-        val result = db.collection(collectionPath).get().await()
-        Log.d("fetchCategories", "Fetched ${result.documents.size} documents")
-        val categories = result.documents.mapNotNull { document -> documentToCategory(document) }
-        withContext(Dispatchers.Main) { onSuccess(categories) }
-      } catch (e: Exception) {
-        Log.e("fetchCategories", "Error fetching categories", e)
-        withContext(Dispatchers.Main) { onFailure(e) }
+    try {
+      // Attempt to fetch from Room first
+      val cachedCategories =
+          dao.getAllCategories()
+              .map { entities -> entities.map { it.toCategory() } }
+              .first() // Take only the first emission from the flow
```

Comment:
Really like this, seeing as a flow returns it continuously, we just take the first output, gjgjgj

Reply:
🐐 🔥